sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -92,67 +94,282 @@ public HttpPipeline getHttpPipeline() {
+     * @return A REST response containing the application/json relationship created.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsResponse<String>> createRelationshipWithResponse(String digitalTwinId, String relationshipId, String relationship) {
+        return withContext(context -> createRelationship(digitalTwinId, relationshipId, relationship, context));
+    }
+
+    Mono<DigitalTwinsResponse<String>> createRelationship(String digitalTwinId, String relationshipId, String relationship, Context context) {
         return protocolLayer
             .getDigitalTwins()
-            .addWithResponseAsync(digitalTwinId, digitalTwin, context)
-            .flatMap(
-                response -> {
-                    T genericResponse = mapper.convertValue(response.getValue(), klazz);
+            .addRelationshipWithResponseAsync(digitalTwinId, relationshipId, relationship, context)
+            .flatMap(response -> {
+                try {
+                    String jsonResponse = mapper.writeValueAsString(response.getValue());
                     DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
-                    return Mono.just(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders));
-                });
+                    return Mono.just(new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), jsonResponse, twinHeaders));
+                } catch (JsonProcessingException e) {
+                    return Mono.error(e);
+                }
+            });
     }
 
     /**
      * Creates a relationship on a digital twin.
      *
      * @param digitalTwinId The Id of the source digital twin.
      * @param relationshipId The Id of the relationship to be created.
-     * @param relationship The application/json relationship to be created.
-     * @return A REST response containing the application/json relationship created.
+     * @param relationship The relationship to be created.
+     * @param clazz The model class to convert the relationship to.
+     * @param <T> The generic type to convert the relationship to.
+     * @return The relationship created.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<String>> createRelationshipWithResponse(String digitalTwinId, String relationshipId, String relationship) {
-        return withContext(context -> createRelationshipWithResponse(digitalTwinId, relationshipId, relationship, context));
+    public <T> Mono<T> createRelationship(String digitalTwinId, String relationshipId, Object relationship, Class<T> clazz) {
+        return withContext(context -> createRelationship(digitalTwinId, relationshipId, relationship, clazz, context))
+            .map(DigitalTwinsResponse::getValue);
```

Comment:
How does this work?

Reply:
This is called a method reference; it is useful when you want to use a lamba expression to call an existing method.

Eg. this `map` operation actually does the following:
```java
.map(new Function<DigitalTwinsResponse<String>, String>() {
        @Override
        public String apply(DigitalTwinsResponse<String> stringDigitalTwinsResponse) {
            return stringDigitalTwinsResponse.getValue();
        }
    });
```

Since all we do in this anonymous class is call `DigitalTwinsResponse.getValue()`, we can replace it by a method reference.